### PR TITLE
feat: add Facebook authentication support

### PR DESCRIPTION
**Pull request by n0mad-d3v**
Hello to the whole Stacked team and thanks again for this nice framework!

In this PR, 
* I added back Facebook authentication via the [flutter_facebook_auth package](https://pub.dev/packages/flutter_facebook_auth) ; 
* I wrote a first documentation on how to use this package ; 
* I created an example application with Google, Facebook and Apple authentication on iOS and Android. The commited example have no secrets left, meaning it's not working out of the box, but here are some videos I did to illustrate the behavior. 

https://github.com/Stacked-Org/firebase_auth/assets/116424218/c35b1a38-ce11-4868-94a5-9aecd806b4e9

Let me know what you think! 🙏

_Christophe_

### CODE REVIEW

**Comment by FilledStacks** [comment]
This is really awesome @n0mad-d3v ! We'll review this and release in this week. 

I appreciate the PR a lot.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 name: stacked_firebase_auth
 description: A service class that provides Firebase Authentication Functionality on a single api
-version: 2.19.6
+version: 2.20.0
```
Comment (ferrarafer) [inline pubspec.yaml:3]: Revert this change, we are using semantic release which takes care of upgrading the version number.

File: pubspec.yaml [hunk 2]
```diff
@@ -20,9 +20,10 @@ dependencies:
   # Firebase Authentications
   google_sign_in: ^6.1.4
   sign_in_with_apple: ^5.0.0
+  flutter_facebook_auth: ^6.0.4
 
   # logging
-  logger: ^1.4.0
+  logger: ^1.1.0
```
Comment (ferrarafer) [inline pubspec.yaml:26]: Which is the conflict we have when we upgrade logger package?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,10 @@
+### Features
+
+* Add implementation of Facebook authentication via `flutter_facebook_auth` dependency
+
+## [2.20.0](https://github.com/Stacked-Org/firebase_auth/compare/v2.19.6...v2.20.0) (2024-02-02)
+
+
```
Comment (ferrarafer) [inline CHANGELOG.md:7]: Please, revert this, semantic version takes care of the changelog.

**Review by ferrarafer — CHANGES_REQUESTED** [review]
Hi @n0mad-d3v, great work ! I'm very happy to see people doing collaboration to the framework. Thanks you very much ❤️ 

Please, do the small changes I requested and we will be ready to merge.

Also, change the name of the PR to **Add Facebook authentication support via `flutter_facebook_auth` dependency** or similar because is what is going to be used by semantic release to describe the change.

Cheers,
Fernando.

File: README.md [hunk 4]
```diff
@@ -1,25 +1,304 @@
...
+    LazySingleton(classType: FirebaseAuthenticationService),
+  ],
+  bottomsheets: [ ... ],
+  dialogs: [ ... ],
+)
+class App {}
+
+```
+
+4. Regrenerate your app via the Stacked CLI: `stacked generate`
```
Comment (ferrarafer) [inline README.md:106]: Typo here, Regrenerate -> Regenerate

File: CHANGELOG.md (hunk 3 above)
Comment (n0mad-d3v) [reply CHANGELOG.md]: Hey @ferrarafer, thanks a lot for the feedback, didn't know!
It's done in the 090d2342d2e5bc91c02cfc1ec6405ed3369af708 ✅

File: README.md (hunk 4 above)
Comment (n0mad-d3v) [reply README.md]: Hey @ferrarafer, indeed!
It's done in the f8aaae611e7c05f98dc8f2ea534f134a35d9dc42 ✅

File: pubspec.yaml (hunk 2 above)
Comment (n0mad-d3v) [reply pubspec.yaml]: Hey @ferrarafer, seems like a mistake indeed. 
I reverted that in the e32f5886f522d91d15cdd379f255f86366446101 ✅

**Review by ferrarafer — CHANGES_REQUESTED** [review]
Great @n0mad-d3v ! Only two more changes and we are ready.

1. Add `firebase_core` dependency to example project.
2. Execute `dart format --fix --set-exit-if-changed .` on the root of the project to update the format to the conventions.
3. Commit changes

**Comment by n0mad-d3v** [comment]
Hey @ferrarafer, 
I did the requested changes 😊
1. `firebase_core` dep in e71872217fa1866bd88d8807c446bb72bca7cc29
2. `dart format --fix --set-exit-if-changed .` in 00cadd4e86da0d6757bd5abaab307d19bff81cec (yes sorry, I'm in `-l 150` by default 😅)

Don't hesitate if there is anything else I may do and thanks for the review 🙏

**Review by ferrarafer — APPROVED** [review]

**Pull request merged**